src/main/java/ch/njol/skript/ScriptLoader.java

```diff
@@ -306,6 +306,37 @@ public static boolean isAsync() {
 		return Collections.unmodifiableCollection(loadedFiles);
 	}
 	
+	/**
+	 * All disabled script files.
+	 */
+	@SuppressWarnings("null")
+	static final Set<File> disabledFiles = Collections.synchronizedSet(new HashSet<>());
+	
+	@SuppressWarnings("null")
+	public static Collection<File> getDisabledFiles() {
+		return Collections.unmodifiableCollection(disabledFiles);
+	}
+
+	/**
+	 * Filter for disabled scripts & folders.
+	 */
+	private final static FileFilter disabledFilter = new FileFilter() {
+		@Override
+		public boolean accept(final @Nullable File f) {
+			return f != null && (f.isDirectory() || StringUtils.endsWithIgnoreCase("" + f.getName(), ".sk")) && f.getName().startsWith("-");
+		}
+	};
+
+	private static void updateDisabledScripts(File directory) {

```

Comment:
shouldn't this clear `disabledFiles`?

Reply:
I mean, maybe. It's only used once really and clearing it would break it because it calls it within to add files within a separate directory.

Reply:
that sounds more like a problem with your implementation, this method doesn't need to be recursive - `Files#walk` handles this beautifully for ya

Reply:
I will update this today :)